bindingcollectionadapter/src/main/java/me/tatarka/bindingcollectionadapter2/itembindings/OnItemBindClass.java

```diff
@@ -105,6 +84,36 @@ public void onItemBind(ItemBinding itemBinding, int position, T item) {
         throw new IllegalArgumentException("Missing class for item " + item);
     }
 
+    public void onItemBindExtra(ItemBinding itemBinding, int position, T item) {
+        SparseArray<Object> extraBindings = null;
+        if (itemExtraBindingMap != null && !itemExtraBindingMap.isEmpty()) {
+            for (int i = 0; i < itemExtraBindingMap.size(); i++) {
+                Class<? extends T> key = itemExtraBindingMap.keyAt(i);
+                if (key.isInstance(item) && extraBindings == null) {
+                    extraBindings = itemExtraBindingMap.valueAt(i);
+                } else {
+                    SparseArray<Object> toRemove = itemExtraBindingMap.valueAt(i);
+                    if (toRemove != null && toRemove.size() > 0) {
+                        for (int j = 0, size = toRemove.size(); j < size; j++) {
+                            int variableId = toRemove.keyAt(j);
+                            itemBinding.removeExtra(variableId);
+                        }
+                    }
+                }
+            }
+            if (extraBindings != null) {
+                for (int j = 0, size = extraBindings.size(); j < size; j++) {
+                    int variableId = extraBindings.keyAt(j);
+                    Object value = extraBindings.valueAt(j);
+                    if (value instanceof OnItemBindClass.PropertyResolver) {
+                        value = ((PropertyResolver<T>) value).resolve(item);
+                    }
+                    itemBinding.bindExtra(variableId, value);
+                }
+            }
+        }
+    }
+
     public interface PropertyResolver<T> {
         Object resolve(T t);
```

Comment:
from your example use case, wouldn't it make sense to pass in the position as well?